model/reliability/query_test.go

```diff
@@ -268,3 +281,376 @@ func TestValidateForTaskReliability(t *testing.T) {
+	docs, err = GetTaskReliabilityScores(filter)
+	require.NoError(err)
+	require.Len(docs, 2)
+	require.NotEqual(docs[0].Date, day1)
+	require.NotEqual(docs[1].Date, day2)
+
+	// filter.StatsFilter.Tasks = []string{task2}
+	// filter.StatsFilter.BuildVariants = []string{variant1, variant2}
+	// filter.StatsFilter.Distros = []string{distro1, distro2}
+	// docs, err = GetTaskReliabilityScores(filter)
+	// require.NoError(err)
+	// require.Len(docs, 2)
+	// for _, doc := range docs {
+	// 	require.Equal(doc.TaskName, task2)
+	// 	require.NotEqual(doc.BuildVariant, variant1)
+	// 	require.NotEqual(doc.Distro, distro1)
+	// }
+
+	// filter.StatsFilter.Tasks = []string{task1}
+	// filter.StatsFilter.Tasks = []string{task1}
+	// filter.StatsFilter.BuildVariants = []string{variant2}
+	// filter.StatsFilter.Distros = []string{distro2}
+	// docs, err = GetTaskReliabilityScores(filter)
+	// require.NoError(err)
+	// require.Len(docs, 0)
+
+	// filter.StatsFilter.Tasks = []string{task2}
+	// filter.StatsFilter.BuildVariants = []string{variant1}
+	// filter.StatsFilter.Distros = []string{distro1}
+	// docs, err = GetTaskReliabilityScores(filter)
+	// require.NoError(err)
+	// require.Len(docs, 0)
+
+	// task3 := "task3"
+	// variantFmt := "variant %04d"
+	// distroFmt := "distro %04d"
+
+	// require.NoError(insertManyDailyTaskStats(MaxQueryLimit, project, "r1", task3, variantFmt, distroFmt, day1, 10, 5, 1, 1, 1, 2, 10.5))
+
+	// filter.StatsFilter.Tasks = []string{task1, task2, task3}
+	// filter.StatsFilter.BuildVariants = []string{}
+	// filter.StatsFilter.Distros = []string{}
+	// docs, err = GetTaskReliabilityScores(filter)
+	// require.NoError(err)
+	// require.Len(docs, MaxQueryLimit)
+}
+
+func TestGroupedResults(t *testing.T) {
+	// after := time.Date(2018, 8, 12, 0, 0, 0, 0, time.UTC)
+	// before := time.Date(2018, 8, 13, 0, 0, 0, 0, time.UTC)
+	requesters := []string{
+		evergreen.PatchVersionRequester,
+		evergreen.GithubPRRequester,
+		evergreen.MergeTestRequester,
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	withCancelledContext := func(ctx context.Context, fn func(context.Context)) {
+		ctx, cancel := context.WithCancel(ctx)
+		cancel()
+		fn(ctx)
+	}
+
+	// filter := createValidFilter()
+	task1 := "task1"
+	task2 := "task2"
+
+	variant1 := "v1"
+	variant2 := "v2"
+
+	distro1 := "d1"
+	distro2 := "d2"
+
+	numSuccess := 20
+	numFailed := 15
+	numTimeout := 1
+	numTestFailed := 2
+	numSystemFailed := 3
+	numSetupFailed := 4
+	avgDuration := 10.5
+
+	for opName, opTests := range map[string]func(ctx context.Context, t *testing.T, env evergreen.Environment){
+		"Grouped": func(ctx context.Context, t *testing.T, env evergreen.Environment) {
+			for testName, testCase := range map[string]func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter){
+				"1 Day / Group 1 ascending": func(ctx context.Context, t *testing.T, filter TaskReliabilityFilter) {
```

Comment:
I think we should probably have seperate subtests and function-like naming for test cases.

Reply:
Renamed containing test function. Subtest by filter field and then added some expected common cases (by day, by week and by month.

This now looks much better / clearer to me.